src/kernel/network.cpp

```diff
@@ -55,211 +147,183 @@ CryptoKernel::Network::Network(CryptoKernel::Log* log,
-
-                        for(const Json::Value& peer : info["peers"]) {
-                            sf::IpAddress addr(peer.asString());
-                            if(addr != sf::IpAddress::None) {
-                                if(!peers->get(dbTx.get(), addr.toString()).isObject()) {
-                                    log->printf(LOG_LEVEL_INFO, "Network(): Discovered new peer: " + addr.toString());
-                                    Json::Value newSeed;
-                                    newSeed["lastseen"] = 0;
-                                    newSeed["height"] = 1;
-                                    newSeed["score"] = 0;
-                                    peers->put(dbTx.get(), addr.toString(), newSeed);
-                                }
-                            } else {
-                                changeScore(it->first, 10);
-                                throw Peer::NetworkError();
-                            }
-                        }
-                    } catch(const Json::Exception& e) {
-                        changeScore(it->first, 50);
-                        throw Peer::NetworkError();
-                    }
-
-                    const std::time_t result = std::time(nullptr);
-                    it->second->info["lastseen"] = static_cast<uint64_t>(result);
-                } catch(const Peer::NetworkError& e) {
-                    log->printf(LOG_LEVEL_WARN,
-                                "Network(): Failed to contact " + it->first + ", disconnecting it");
-                    removals.insert(it->first);
-                }
-            }
-
-            for(const auto& peer : removals) {
-                const auto it = connected.find(peer);
-                peers->put(dbTx.get(), peer, it->second->info);
-                if(it != connected.end()) {
-                    connected.erase(it);
-                }
-            }
-
-            for(const auto& peer : peerInfos) {
-                peers->put(dbTx.get(), peer.first, peer.second);
-            }
+void CryptoKernel::Network::makeOutgoingConnectionsWrapper() {
+	while(running) {
+		makeOutgoingConnections();
+		std::this_thread::sleep_for(std::chrono::milliseconds(100));
+	}
+}
 
-            dbTx->commit();
+void CryptoKernel::Network::infoOutgoingConnectionsWrapper() {
+	while(running) {
+		bool wait = false;
 
-            std::lock_guard<std::mutex> cLock(connectedStatsMutex);
-            connectedStats.clear();
+		infoOutgoingConnections();
+		std::this_thread::sleep_for(std::chrono::milliseconds(100));
+	}
+}
 
-            for(const auto& peer : connected) {
-                peerStats stats = peer.second->peer->getPeerStats();
-                stats.version = peer.second->info["version"].asString();
-                stats.blockHeight = peer.second->info["height"].asUInt64();
-                connectedStats.insert(std::make_pair(peer.first, stats));
```

Comment:
`connectedStats` map is never filled with anything so `getpeerinfo` always returns null